### PR TITLE
File upload script

### PR DESCRIPTION
**Pull request by imdfl**
$ `yarn upload <path> [path ...] [--tags tag1 [tag2...]]`

scripts/.env file in private message

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,19 +50,23 @@
 		"@radix-ui/react-tooltip": "1.0.2",
 		"@sendgrid/mail": "^7.7.0",
 		"@stitches/react": "^1.2.8",
+		"aws": "^0.0.3-2",
```
Comment (tomerlichtash) [inline package.json:53]:
```
		"aws": "^0.0.3-2",
		"aws-sdk": "^2.1399.0",
```
Can this be a Dev Dependency instead of a regular dependency?

**Review by tomerlichtash — CHANGES_REQUESTED** [review]
I don't have the script to upload my appreciation for this. Excellent stuff.
Added some minor type/infra comments.

Will check out the lint issue.
Also, for some reason build doesn't run properly (not because of lint).

File: tsconfig.json [hunk 2]
```diff
@@ -19,8 +19,10 @@
 	"include": [
 		"next-env.d.ts",
 		"globals.d.ts",
-		"**/**/*.ts",
-		"**/**/*.tsx",
+		"src/*.ts",
```
[the remark this answers is not in the document]
Comment (imdfl) [reply tsconfig.json]: Reverted, added exclude

File: scripts/s3-client.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+		return process.env.AWS_BUCKET || "";
+	}
+
+	public get region() {
+		return process.env.AWS_REGION || "";
+	}
+
+	public get client(): S3Client {
+		if (!this._client) {
+			this._client = new (S3Client as any)({
```
[the remark this answers is not in the document]
Comment (imdfl) [reply scripts/s3-client.ts:46]: tsc doesn't recognize the constructor even tho it knows the class. Odd, but I didn't investigate.

File: scripts/list-media.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        Bucket: proxy.bucket,
+
+    });
+    const response = await proxy.client.send(cmd);
+    return response;
+}
+
+
+
+listObjects().then((response: any) => {
```
[the remark this answers is not in the document]
Comment (imdfl) [reply scripts/list-media.ts]: typed, but typescript is the default, it's the path of least effort, so if any happens to gets things done, it's not antithetic to the platform.

File: scripts/list-media.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+
```
[the remark this answers is not in the document]
Comment (imdfl) [reply scripts/list-media.ts:1]: fixed

**Comment by imdfl** [comment]
@tomerlichtash , please review

File: scripts/package.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  "version": "1.0.0",
+  "description": "Scripts for Mel's Loop",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "dfl",
+  "license": "MIT",
+  "dependencies": {
+    "@aws-sdk/client-s3": "^3.354.0",
```
[the remark this answers is not in the document]
Comment (imdfl) [reply scripts/package.json:12]: fixed

File: scripts/s3-client.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (tomerlichtash) [reply scripts/s3-client.ts:46]: Try just to remove `any`:

```
			this._client = new S3Client({
				region: this.region,
			});
```

On my machine it resolved fine, and referring to the type.

File: scripts/tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,65 @@
+{
+	"compilerOptions": {
+		/* Visit https://aka.ms/tsconfig.json to read more about this file */
+		/* Basic Options */
+		// "incremental": true,                   /* Enable incremental compilation */
+		"target": "ES2020", /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */
+		"module": "commonjs", /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', 'es2020', or 'ESNext'. */
+		// "lib": [],                             /* Specify library files to be included in the compilation. */
```
Comment (tomerlichtash) [inline scripts/tsconfig.json:8]: Let's keep only what we need in this tsconfig. Unless you want that configuration for later.

**Comment by tomerlichtash** [comment]
@imdfl - Can you please also add an _EMPTY_ `.env` file to the scripts folder?

```
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_BUCKET=
AWS_REGION=
```

**Comment by tomerlichtash** [comment]
> @imdfl - Can you please also add an _EMPTY_ `.env` file to the scripts folder?
> 
> ```
> AWS_ACCESS_KEY_ID=
> AWS_SECRET_ACCESS_KEY=
> AWS_BUCKET=
> AWS_REGION=
> ```
> 
> After comitting this file, add `*.env` to `.gitignore`, to prevent an accidental push.

Please disregard for now. We'll try vercel's CLI which has a method to pull env vars.

**Comment by imdfl** [comment]
How do you envision the upload script running on a vercel machine?

I can extend the script so that it accepts s3 urls in addition to local paths. Then you can upload a file from one bucket to another. Later I can add a script to upload a local file to the s3 bucket from which the final upload will be performed. That script will require different env variables, no biggie. To allow the second script to run on vercel, we will need to support uploading from a third bucket, but this is already solved.

**Comment by tomerlichtash** [comment]
> How do you envision the upload script running on a vercel machine?

I merely referred to the idea of a secret manager, that we get from Vercel. I'm not sure if this is relevant without a Pro Plan, otherwise no one but me would be able to access it, but generally the idea is to run a CLI that will clone an .env file to your dev env. Perhaps this can be achieved with another service, researching it now.

> I can extend the script so that it accepts s3 urls in addition to local paths. Then you can upload a file from one bucket to another. Later I can add a script to upload a local file to the s3 bucket from which the final upload will be performed. That script will require different env variables, no biggie. To allow the second script to run on vercel, we will need to support uploading from a third bucket, but this is already solved.

Not sure I follow the use case here.

File: scripts/s3-client.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (imdfl) [reply scripts/s3-client.ts]: no idea what has changed, but it compiles fine now.

**Review by tomerlichtash — APPROVED** [review]

**Pull request merged**